galasa-ui/src/middleware.ts

```diff
@@ -8,16 +8,25 @@ import { NextResponse, NextRequest } from 'next/server';
 import AuthCookies from './utils/authCookies';
 import { GALASA_WEBUI_CLIENT_ID, authApiClient, sendAuthRequest } from './utils/auth';
 import { AuthProperties } from './generated/galasaapi';
+import { cookies } from 'next/headers';
 
 // Runs before any request is completed
 export async function middleware(request: NextRequest) {
   let response = NextResponse.rewrite(new URL('/error', request.url));
 
   try {
     if (request.url.includes('/callback')) {
-      // Handle callback requests
       const responseUrl = request.url.substring(0, request.url.lastIndexOf('/callback'));
-      response = await handleCallback(request, NextResponse.redirect(responseUrl, { status: 302 }));
+
+      const shouldReturnToMySettingsPage = cookies().get(AuthCookies.SHOULD_REDIRECT_TO_SETTINGS)
+    
+      if(shouldReturnToMySettingsPage?.value === 'true'){
+        response = await handleCallback(request, NextResponse.redirect(responseUrl + "/mysettings", { status: 302 }));  
+      }
+      else{     
+        response = await handleCallback(request, NextResponse.redirect(responseUrl, { status: 302 }));
+      }
```

Comment:
Instead of duplicating the callback handling function call - you can turn the responseUrl into a variable and update that